### PR TITLE
Fix last super spec

### PR DESCRIPTION
**Pull request by chrisseaton**
Progress towards #2103.

https://github.com/Shopify/truffleruby/issues/1.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Compatibility:
 * Verify that gem paths are correct before loading RubyGems (#2075).
 * Implement `rb_ivar_count`.
 * Implemented `rb_yield_values2`.
+* Pass the final `super` specs (#2104, #chrisseaton).
```
Comment (eregon) [inline CHANGELOG.md:35]:
```suggestion
* Pass the final `super` specs (#2104, @chrisseaton).
```

File: src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java [hunk 2]
```diff
@@ -48,7 +48,15 @@ public final Object execute(VirtualFrame frame) {
 
         if (restArgIndex != -1) {
             final Object restArg = superArguments[restArgIndex];
-            final Object[] restArgs = unsplat((RubyArray) restArg);
+
+            final Object[] restArgs;
+
+            if (restArg instanceof RubyArray) {
```
Comment (eregon) [inline src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java:54]: Should we profile this condition (with ConditionProfile)?

Comment (chrisseaton) [reply src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java:54]: Could do - but I'll be doing a check of the performance of `super` after the specs pass so I'll update it then.

Comment (eregon) [reply src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java:54]: I'd rather do it eagerly if you don't mind.
I think in general we currently profile every condition unless it folds at PE time or known to be worthless to profile.
I can also do it when integrating the PR.

**Review by eregon — APPROVED** [review]

File: src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java (hunk 2 above)
Comment (chrisseaton) [reply src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java:54]: Sure you can do it then - there's a couple of unprofiled branches in this method then.

Comment (eregon) [reply src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java:54]: Which other branches? `restArgIndex` is PE constant.

**Pull request merged**